src/service-override/views.ts

```diff
@@ -207,9 +227,10 @@ interface SimpleEditorInput extends EditorInput {
   setDirty (dirty: boolean): void
 }
 
-function registerEditorPane (options: EditorPanelOption): { disposable: IDisposable, CustomEditorInput: new (closeHandler?: IEditorCloseHandler) => SimpleEditorInput } {
+function registerEditorPane (options: EditorPanelOption): { disposable: IDisposable, CustomEditorInput: new (closeHandler?: IEditorCloseHandler, baseInput?: IResourceEditorInput | ITextResourceEditorInput) => SimpleEditorInput } {
```

Comment:
doesn't `ITextResourceEditorInput` extends `IResourceEditorInput`?

Reply:
Right, I've taken this from the editor input factory that needs to be passed to the editor resolver:

<img width="1400" alt="image" src="https://github.com/CodinGame/monaco-vscode-api/assets/587016/72e167cd-9081-4478-b1f4-2aad91593d1c">

Which is used here and passed as `baseInput`:

<img width="1220" alt="image" src="https://github.com/CodinGame/monaco-vscode-api/assets/587016/cc6e0407-664f-43d1-a4e0-61a83af79b20">

I can change it to only use `IResourceEditorInput`, but that would make it a bit less consistent with the API of VSCode.

Reply:
I do agree that we can just accept `IResourceEditorInput`. The only reason to keep the union is to indicate that this is the type that VSCode passes to the `createEditorInput`.